drivers/SmartThings/zigbee-fan/src/kichler-fan-light/init.lua

```diff
@@ -0,0 +1,187 @@
+-- Copyright 2024 SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+local clusters = require "st.zigbee.zcl.clusters"
+local capabilities = require "st.capabilities"
+local FanControl = clusters.FanControl
+local Level = clusters.Level
+local OnOff = clusters.OnOff
+
+local FINGERPRINTS = {
+  { mfr = "KICHLER", model = "KICHLER-FANLIGHT-Z-301" },
+}
+
+local function can_handle_itm_fanlight(opts, driver, device)
+  for _, fingerprint in ipairs(FINGERPRINTS) do
+    if device:get_manufacturer() == fingerprint.mfr and device:get_model() == fingerprint.model then
+      return true
+    end
+  end
+  return false
+end
+
+-- CAPABILITY HANDLERS
+local function on_handler(driver, device, command)
+  local last_level = device:get_field('LAST_DIM_LEVEL') or 100
```

Comment:
as far as I can tell, you maintain the `level` value even when receiving an `off` report. Unless the device separately reports a `0` level when turning off, you should be able to use `device:get_latest_state` instead of setting an additional field here.

Reply:
When i apply the code that was guided to me, if i turn on the light after turning it off, i can't retrieve the last value (off), and instead retrieve the off state. so,  I will continue to use the set/get_field as you are now.

Reply:
were you getting the latest value of the `switch` capability or the `switchLevel` capability?

```
device:get_latest_state("main", capabilities.switchLevel.ID, capabilities.switchLevel.level.NAME)
```

Reply:
The use of the switchlevel is because it is synchronized with the main switch.
The proposed get_lastest_state maintains "off" if you turn it on after turning it off, as the last state value is "off".
Therefore, we intend to use set/get field instead of using get_lastest_state.